api/src/main/java/org/pmiops/workbench/api/ProfileController.java

```diff
@@ -269,11 +273,20 @@ private User initializeUserIfNeeded() {
         return user;
 
       case ERROR:
-        log.log(Level.SEVERE, String.format(
-            "free tier project %s failed to be created", user.getFreeTierBillingProjectName()));
-        user.setFreeTierBillingProjectStatus(status);
-        return userDao.save(user);
-
+        int retries = Optional.ofNullable(user.getBillingProjectRetries()).orElse(0);
+        if (retries <= 2) {
+          this.userService.setBillingRetryCount(retries + 1);
+          fireCloudService.removeUserFromBillingProject(user.getEmail(), user.getFreeTierBillingProjectName());
+          String billingProjectName = createFirecloudBillingProject(user);
+          user.setFreeTierBillingProjectName(billingProjectName);
+          user.setFreeTierBillingProjectStatus(BillingProjectStatus.PENDING);
+          return userDao.save(user);
```

Comment:
Again, better to use UserService to save this stuff with retries, so we handle race conditions appropriately.

Reply:
Will do. This is what we were doing for saving user billing project stuff before. Should I move that over too?